src/System.Text.Json/tests/Serialization/SnakeCaseUnitTests.cs

```diff
@@ -0,0 +1,141 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static class SnakeCaseUnitTests
+    {
+        [Fact]
+        public static void Convert_SpecifiedName_MatchesExpectedName()
+        {
+            Assert.Equal("url_value", ConvertToSnakeCase("URLValue"));
+            Assert.Equal("url", ConvertToSnakeCase("URL"));
+            Assert.Equal("id", ConvertToSnakeCase("ID"));
+            Assert.Equal("i", ConvertToSnakeCase("I"));
+            Assert.Equal("", ConvertToSnakeCase(""));
+            Assert.Null(ConvertToSnakeCase(null));
+            Assert.Equal("person", ConvertToSnakeCase("Person"));
+            Assert.Equal("i_phone", ConvertToSnakeCase("iPhone"));
+            Assert.Equal("i_phone", ConvertToSnakeCase("IPhone"));
+            Assert.Equal("i_phone", ConvertToSnakeCase("I Phone"));
+            Assert.Equal("i_phone", ConvertToSnakeCase("I  Phone"));
+            Assert.Equal("i_phone", ConvertToSnakeCase(" IPhone"));
+            Assert.Equal("i_phone", ConvertToSnakeCase(" IPhone "));
+            Assert.Equal("is_cia", ConvertToSnakeCase("IsCIA"));
+            Assert.Equal("vm_q", ConvertToSnakeCase("VmQ"));
+            Assert.Equal("xml2json", ConvertToSnakeCase("Xml2Json"));
+            Assert.Equal("sn_ak_ec_as_e", ConvertToSnakeCase("SnAkEcAsE"));
+            Assert.Equal("sn_a__k_ec_as_e", ConvertToSnakeCase("SnA__kEcAsE"));
+            Assert.Equal("sn_a__k_ec_as_e", ConvertToSnakeCase("SnA__ kEcAsE"));
+            Assert.Equal("already_snake_case_", ConvertToSnakeCase("already_snake_case_ "));
+            Assert.Equal("is_json_property", ConvertToSnakeCase("IsJSONProperty"));
+            Assert.Equal("shouting_case", ConvertToSnakeCase("SHOUTING_CASE"));
+            Assert.Equal("9999_12_31t23_59_59_9999999z", ConvertToSnakeCase("9999-12-31T23:59:59.9999999Z"));
+            Assert.Equal("hi_this_is_text_time_to_test", ConvertToSnakeCase("Hi!! This is text. Time to test."));
+            Assert.Equal("building", ConvertToSnakeCase("BUILDING"));
+            Assert.Equal("building_property", ConvertToSnakeCase("BUILDING Property"));
+            Assert.Equal("building_property", ConvertToSnakeCase("Building Property"));
+            Assert.Equal("building_property", ConvertToSnakeCase("BUILDING PROPERTY"));
+        }
```

Comment:
Nit: can this instead be:
```C#
[Theory]
[InputData(..., ...)] // the cases you have
public static void Convert_SpecifiedName_MatchesExpectedName(string expected, string input)
{
    Assert.Equal(expected, JsonNamingPolicy.SnakeCase.ConvertName(name));
}
```

Reply:
Yep, but I assumed that @ahsonkhan would like to implement it this way to compare behaviors. Are you okay with changing this, @ahsonkhan?

Reply:
Yep, the change seems reasonable to me. Tests showing the expected/actual string before/after is clear in both the current way the test is written and what @stephentoub suggested, so I am supportive of the the change.